transformer_engine/pytorch/attention.py

```diff
@@ -5013,7 +5015,7 @@ def forward(
                 if _flash_attn_2_4_1_plus:
                     fa_optional_forward_kwargs["deterministic"] = self.deterministic
                 fa_optional_forward_args_thd = []
-                if qkv_format in ["bshd", "sbhd"] and "padding" not in attn_mask_type:
+                if (qkv_format in ["bshd", "sbhd"] and "padding" not in attn_mask_type) or fp8:
```

Comment:
if fp8 is true, I think `qkv_format in ["bshd", "sbhd"] and "padding" not in attn_mask_type` is supposed to be true for sure? so adding `or fp8` seems redundant?